### PR TITLE
PR: Correctly set namespace while debugging

### PR DESCRIPTION
**Pull request by impact27**
`runfile` and `runcell` can be used while debugging to run code. However, the namespace is not properly set. This PR solves this problem.

### CODE REVIEW

File: spyder_kernels/customize/namespace_manager.py [hunk 1]
```diff
@@ -12,10 +12,22 @@
 from spyder_kernels.py3compat import PY2
 
 
-def _get_globals():
-    """Return current namespace"""
-    ipython_shell = get_ipython()
-    return ipython_shell.user_ns
+def _get_globals_locals():
```
Comment (ccordoba12) [inline spyder_kernels/customize/namespace_manager.py]: We already have a method called `_get_current_namespace` in SpyderKernel, which is almost exactly as this one. I think it'd be better to use that instead.

File: spyder_kernels/customize/namespace_manager.py [hunk 2]
```diff
@@ -12,10 +12,22 @@
...
+        pdb = get_ipython().kernel._pdb_obj
+        ns_locals = pdb.curframe_locals
+        ns_globals = pdb.curframe.f_globals
+    else:
+        ns_locals = None
+        ns_globals = get_ipython().user_ns
+    return ns_globals, ns_locals
+
+
+def _set_globals_locals(ns_globals, ns_locals):
```
Comment (ccordoba12) [inline spyder_kernels/customize/namespace_manager.py]: Why does this function need an `ns_locals` arg that it's not using?

File: spyder_kernels/customize/namespace_manager.py [hunk 3]
```diff
@@ -29,7 +41,7 @@ class NamespaceManager(object):
     def __init__(self, filename, namespace=None, current_namespace=False,
                  file_code=None):
         self.filename = filename
-        self.namespace = namespace
+        self.namespace = namespace, None
```
Comment (ccordoba12) [inline spyder_kernels/customize/namespace_manager.py]: Why do you need to add `None` here? That's really not clear to me.

Comment (ccordoba12) [reply spyder_kernels/customize/namespace_manager.py]: It seems you're using `self.namespace` to refer to `globals` and `locals`. If that's the case, please use `self.ns_globals` and `self.ns_locals` instead.

File: spyder_kernels/customize/namespace_manager.py (hunk 1 above)
Comment (impact27) [reply spyder_kernels/customize/namespace_manager.py]: I want to get both global and local separately as they are different arguments to `exec`. `_get_current_namespace` returns a mix of the two in a single dictionnary.

**Pull request merged**